src/spatial/hop_rates.jl

```diff
@@ -57,30 +57,22 @@ function HopRates(p::Pair{SpecHop, SiteHop},
 end
 
 """
-    update_hop_rates!(hop_rates::AbstractHopRates, species::AbstractArray, u, site, spatial_system)
+    update_hop_rates!(hop_rates::AbstractHopRates, species_vec, u, site, spatial_system)
 
-update rates of all specs in species at site
+update rates of all species in species_vec at site
 """
-function update_hop_rates!(hop_rates::AbstractHopRates, species::AbstractArray, u, site,
-                           spatial_system)
-    @inbounds for spec in species
-        update_hop_rate!(hop_rates, spec, u, site, spatial_system)
+function update_hop_rates!(hop_rates::AbstractHopRates, species_vec, u, site, spatial_system)
```

Comment:
If `species_vec` is a single species, the function reduces to updating a single hop rate. So I deleted `update_hop_rate!`.

Reply:
Or are you saying this is always called with a vector from higher levels so it would still have the vector overhead?

Reply:
I don't remember my reasoning now. I put it back in.